django/template/base.py

```diff
@@ -841,12 +840,12 @@ def __repr__(self):
     def render(self, context):
         try:
             output = self.filter_expression.resolve(context)
+            return render_value_in_context(output, context)
         except UnicodeDecodeError:
             # Unicode conversion can fail sometimes for reasons out of our
             # control (e.g. exception rendering). In that case, we fail
             # quietly.
             return ''
-        return render_value_in_context(output, context)
```

Comment:
In the debug tests the following string is passed to a template and used as an argument to a template tag:

```
b'\xa1Broken!'  # non-ASCII, latin1 bytestring
```

This raises a ``UnicodeDecodeError`` when ``force_text`` is called from ``render_value_in_context``. The old ``DebugVariableNode`` passed this test because it didn't use ``render_value_in_context``. Instead it called ``force_text``, but inside a ``try: ... except UnicodeDecodeError`` block.

It's not clear to me if that difference in behavior was on purpose. This is a case where it would fail silently in debug and fail loudly when not.

My inclination is to undo moving ``render_value_in_context``, change the test, and not silence the exception when the result is a string that can't be converted to unicode.

Thoughts?